18-websockets/classworks/Agatha Rudko/task/src/pages/Chat/Chat.jsx

```diff
@@ -0,0 +1,91 @@
+import React, {useState, useEffect} from 'react';
+import './Chat.css';
+import Message from './components/Message';
+import { socket } from "../../socket";
+
+function Chat(props) {
+  const [value, setValue] = useState('');
+  const [name, setName] = useState('');
+  const [isTyping, setIsTyping] = useState(false);
+  const [typingUser, setTypingUser] = useState('')
+  const [messages, setMessages] = useState([]);
+
+  socket.on('render_message', data => {
+    setMessages([...messages, data]);
+  });
+
+  if(isTyping){
+    socket.emit('typing', name);
```

Comment:
Looks like you will send a lot of requests to the server
A good practice is to use the `debounce` method for calling function no more than one time per 2-4 secs.
Unfortunately, javascript doesn't have this method. You can write it by yourself or use some library.